test/cli_run_dns_test.go

```diff
@@ -66,3 +68,72 @@ func (suite *PouchRunSuite) TestRunWithBridgeNetwork(c *check.C) {
 
 	c.Assert(res.Stdout(), check.Equals, hostRes.Stdout())
 }
+
+// TestRunWithDNSFlags tests DNS related flags.
+func (suite *PouchRunDNSSuite) TestRunWithDNSFlags(c *check.C) {
+	cname := "TestRunWithDNSFlags"
+
+	res := command.PouchRun("run", "--name", cname,
+		"--dns", "1.2.3.4",
+		"--dns-option", "timeout:3",
+		"--dns-search", "example.com",
+		busyboxImage,
+		"cat", "/etc/resolv.conf")
+	defer DelContainerForceMultyTime(c, cname)
+	res.Assert(c, icmd.Success)
+
+	// test if the value is in inspect result
+	inspectRes := command.PouchRun("inspect", cname)
+	res.Assert(c, icmd.Success)
+
+	result := []types.ContainerJSON{}
```

Comment:
I add a util func named inspectFilter in pr #2372 , use it to instead of unmarshal?

Reply:
Yeah, please rebase this pr to the latest master. And do as @ZYecho said.

Reply:
Rebased. And fixed as ZYecho said. @allencloud